vllm/v1/sample/ops/topk_topp_sampler.py

```diff
@@ -28,9 +29,12 @@ class TopKTopPSampler(nn.Module):
     Implementations may update the logits tensor in-place.
     """
 
-    def __init__(self):
+    def __init__(self, logprobs_mode: LogprobsMode = "raw_logprobs") -> None:
         super().__init__()
-        if current_platform.is_cuda():
+        self.logprobs_mode = logprobs_mode
```

Comment:
It would be better here to not store/compare a string, just have an enum with three values.